Fixing pyre issues

YLGH opened the pull request:
Summary: refactor and address some pyre issues

Differential Revision: D42905705

colin2328 commented on .github/workflows/pyre.yml:
```diff
@@ -0,0 +1,28 @@
+        run: >
+          conda install --yes -c pytorch-nightly "pytorch-nightly"::pytorch[build="*cpu*"] &&
+          pip install -r requirements.txt &&
+          pip uninstall fbgemm_gpu-nightly -y &&
```
rather than do this, we can replace with CPU: sed -i 's/fbgemm-gpu-nightly/fbgemm-gpu-nightly-cpu/g' requirements.txt

YLGH replied:
sure, but is there a benefit to doing this?

YLGH replied:
ah.- I guess it saves an install

Commit:
Fixing pyre issues and enable OSS Pyre. Remove numpy as TorchRec dependency (#1003)

Summary:
Pull Request resolved: https://github.com/pytorch/torchrec/pull/1003

refactor and address some pyre issues

There are issues with numpy.
Internally we're fixed to version 1.19.5, however, this does not work in OSS since it is incompatible with python 3.10. Numpy is only used in datasets - which we now ignore for pyre externally.

Reviewed By: MichaelCuevas

Differential Revision: D42905705

fbshipit-source-id: 63c7927d3fa65ae13057c431190c1e9300442279
